File: 000_C# Training/C#_2 Classes and methods/Ruby/MainClass.rb
def MethodInTheSameClass
    puts "Hello, i am the method in the same class"
end

def MethodReturningValue
    puts "Hello, I return a string. I automatically return the value from the last line, or everything after the keyword 'return'"
    returnValue = "ABC"
    # or return "ABC"
end

def MethodReceivingValue(receivedString)
    puts "Hello, I received the string " + receivedString + "."
end

class OtherClass
    def self.MethodInAnotherClass()
        puts "Hello, I am the static method in another class"
    end
end
    
MethodInTheSameClass()
OtherClass.MethodInAnotherClass()
returnValue = MethodReturningValue()
puts returnValue

MethodReceivingValue("XYZ")

gets.chomp

File: 000_C# Training/C#_2 Classes and methods/Java/MainClass.java
import java.util.Scanner;

class MainClass
{
    public static void main(String[] args)
    {
        MethodInTheSameClass();
        OtherClass.MethodInAnotherClass();
        String returnValue = MethodReturningValue();
        System.out.println(returnValue);
        
        MethodReceivingValue("XYZ");

        new Scanner(System.in).nextLine();
    }

    public static void MethodInTheSameClass()
    {
        System.out.println("Hello, i am the method in the same class. I am void meaning that I don't return anything. My VBA counterpart would be 'Sub'");
    }

    public static String MethodReturningValue()
    {
        System.out.println("Hello, I return a string. So I need at least one time the key word 'return' followed by the string that I return. My VBA counterpart would be 'Function'");
        return "ABC";
    }

    public static void MethodReceivingValue(String receivedString)
    {
        System.out.println("Hello, I received the string " + receivedString + ".");
    }   
}